keras_cv/layers/preprocessing/random_hue.py

```diff
@@ -28,40 +28,45 @@ class RandomHue(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
     hue channel (H) by delta. The image is then converted back to RGB.
 
     Args:
-        factor: Either a tuple of two floats or a single float. `factor` controls the
-            extent to which the image saturation is impacted. `factor` =
-            `0.0`, `0.5` or `1.0` makes this layer perform a no-op operation.
+        factor: A tuple of two floats, a single float or `keras_cv.FactorSampler`.
+            factor` controls the extent to which the image saturation is impacted.
+            `factor`=`0.0`, `0.5` or `1.0` makes this layer perform a no-op operation.
             `factor=0.25` and `factor=0.75` makes the image to have fully opposite
             hue value. Values should be between `0.0` and `1.0`.
             If a tuple is used, a `factor` is sampled
             between the two values for every image augmented.  If a single float is
             used, a value between `0.0` and the passed float is sampled.
             In order to ensure the value is always the same, please pass a tuple with
             two identical floats: `(0.5, 0.5)`.
+        value_range:  the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.  Defaults to
+            `[0, 255]`.
     """
 
-    def __init__(self, factor, **kwargs):
+    def __init__(self, factor, value_range, **kwargs):
         super().__init__(**kwargs)
-        self.factor = preprocessing.parse_factor_value_range(
-            factor, min_value=0.0, max_value=1.0
+        self.factor = preprocessing.parse_factor(
+            factor,
         )
+        self.value_range = value_range
 
     def get_random_transformation(self, image=None, label=None, bounding_box=None):
         del image, label, bounding_box
-        if self.factor[0] == self.factor[1]:
-            return self.factor[0]
-        return self._random_generator.random_uniform(
-            shape=(), minval=self.factor[0], maxval=self.factor[1], dtype=tf.float32
-        )
+        invert = preprocessing.random_inversion(self._random_generator)
```

Comment:
I am bit confused here, so there are two randomness here? (one for +-1 and another from self.factor()?). Is this what we expect here?

Reply:
I do think this is what we want, because hue adjustments can be in range [-1, 1], but we want end users to only pass [0, 1] so that they can think about magnitudes instead of thinking about transformed values.  Before, it was a worse UX to pass (0.5 0.5) and get a no-op.  

While this is slightly less performant, I believe it affords us a much cleaner end user API.

Reply:
Note that when user specify (1.0, 1.0), the value we send to tf.image.adjust_hue() is either -1 or 1. In that case, the output is actually same as the input. To achieve the max hue adjustment, the delta should be either -0.5 or 0.5 I think.